main.tf

```diff
@@ -25,11 +25,14 @@ resource "google_dns_managed_zone" "peering" {
   visibility    = "private"
   force_destroy = var.force_destroy
 
-  private_visibility_config {
-    dynamic "networks" {
-      for_each = var.private_visibility_config_networks
-      content {
-        network_url = networks.value
+  dynamic "private_visibility_config" {
+    for_each = length(var.private_visibility_config_networks) > 1 ? [1] : []
```

Comment:
Length should be > 0 for this to execute.
length(var.private_visibility_config_networks) > 0